app/src/main/java/org/openobservatory/ooniprobe/domain/GetTestSuite.java

```diff
@@ -53,4 +53,28 @@ public AbstractSuite getFrom(Result result) {
 
         return testSuite;
     }
+
+    public AbstractSuite getFrom(Result result, List<String> inputs) {
+        AbstractSuite testSuite = result.getTestSuite();
+        WebConnectivity test = new WebConnectivity();
+
+        // possible NPE from measurements whose url's are null.
+        List<Url> urls = Lists.transform(
+                Lists.newArrayList(
+                        Iterables.filter(result.getMeasurements(), input -> input.url != null)
```

Comment:
I think here we should _additionally_ filter by the `Measurement.test_name` and ensure we only run this code for Web Connectivity measurements. Because we are also doing something similar in the menu, this code will only be introduced for robustness (which makes sense given that code in different files and packages is likely to diverge sooner or later). As to why I am advocating for this change, the reason is again robustness. Not only we want to avoid trying to re-run experiments with null URLs (which you correctly checked for), but also we'd like to be safe in case we introduce new experiments with non-null URLs in the future.